lib/compat/wordpress-6.0/class-wp-theme-json-gutenberg.php

```diff
@@ -159,7 +171,37 @@ public function get_data() {
 			}
 		}
 
+		$flattened_theme_json = static::do_opt_out_of_settings( $flattened_theme_json );
+
 		return $flattened_theme_json;
 	}
 
+	protected static function do_opt_in_into_settings( &$context ) {
+		foreach ( static::TO_OPT_IN as $path ) {
+			// Use "unset prop" as a marker instead of "null" because
+			// "null" can be a valid value for some props (e.g. blockGap).
+			if ( 'unset prop' === _wp_array_get( $context, $path, 'unset prop' ) ) {
+				_wp_array_set( $context, $path, true );
+			}
+		}
+	}
+
+	protected static function do_opt_out_of_settings( $theme_json ) {
+		if ( array_key_exists( 'appearanceTools', $theme_json['settings'] ) ) { // Is it safe to assume that 'settings' always exsits?
+			foreach ( static::TO_OPT_IN as $path ) {
+				// Remove the path.
+				if ( ! empty( $theme_json['settings'][ $path[ 0 ] ][ $path[ 1 ] ] ) ) {
```

Comment:
This is really hard to read because of the $path[ 0 ] thing, but I'm not sure what would be a better approach...

Reply:
Maybe something like

```php
foreach ( static::TO_OPT_IN as list( $group, $prop ) ) {
	if ( ! empty( $theme_json['settings'][ $group ][ $prop ] ) ) {
```